components/generators/kubernetes/__init__.py

```diff
@@ -72,7 +72,7 @@ def add_volumes_for_objects(self, objects):
             self.root.spec.template.spec.volumes += [{
                 'name': object_name,
                 key: {
-                    'defaultMode': object.root.spec.get('default_mode', 420),
+                    "defaultMode": object.config.get('default_mode', 420),
```

Comment:
have you installed the pre-commit hooks?
You're now mixing double and single quotes